examples/ensemble/plot_stack_predictors.py

```diff
@@ -188,62 +188,40 @@ def load_ames_housing():
 # Now we can use Ames Housing dataset to make the predictions. We check the
 # performance of each individual predictor as well as of the stack of the
 # regressors.
-#
-# The function ``plot_regression_results`` is used to plot the predicted and
-# true targets.
 
 
 import time
 import matplotlib.pyplot as plt
+from sklearn.metrics import PredictionErrorDisplay
 from sklearn.model_selection import cross_validate, cross_val_predict
 
-
-def plot_regression_results(ax, y_true, y_pred, title, scores, elapsed_time):
-    """Scatter plot of the predicted vs true targets."""
-    ax.plot([y_true.min(), y_true.max()],
-            [y_true.min(), y_true.max()],
-            '--r', linewidth=2)
-    ax.scatter(y_true, y_pred, alpha=0.2)
-
-    ax.spines['top'].set_visible(False)
-    ax.spines['right'].set_visible(False)
-    ax.get_xaxis().tick_bottom()
-    ax.get_yaxis().tick_left()
-    ax.spines['left'].set_position(('outward', 10))
-    ax.spines['bottom'].set_position(('outward', 10))
-    ax.set_xlim([y_true.min(), y_true.max()])
-    ax.set_ylim([y_true.min(), y_true.max()])
-    ax.set_xlabel('Measured')
-    ax.set_ylabel('Predicted')
-    extra = plt.Rectangle((0, 0), 0, 0, fc="w", fill=False,
-                          edgecolor='none', linewidth=0)
-    ax.legend([extra], [scores], loc='upper left')
-    title = title + '\n Evaluation in {:.2f} seconds'.format(elapsed_time)
-    ax.set_title(title)
-
-
 fig, axs = plt.subplots(2, 2, figsize=(9, 7))
 axs = np.ravel(axs)
 
 for ax, (name, est) in zip(axs, estimators + [('Stacking Regressor',
                                                stacking_regressor)]):
+    scorers = {"R2": "r2", "MAE": "neg_mean_absolute_error"}
+
     start_time = time.time()
-    score = cross_validate(est, X, y,
-                           scoring=['r2', 'neg_mean_absolute_error'],
-                           n_jobs=-1, verbose=0)
+    scores = cross_validate(
+        est, X, y, scoring=list(scorers.values()), n_jobs=-1, verbose=0
+    )
     elapsed_time = time.time() - start_time
 
     y_pred = cross_val_predict(est, X, y, n_jobs=-1, verbose=0)
-
-    plot_regression_results(
-        ax, y, y_pred,
-        name,
-        (r'$R^2={:.2f} \pm {:.2f}$' + '\n' + r'$MAE={:.2f} \pm {:.2f}$')
-        .format(np.mean(score['test_r2']),
-                np.std(score['test_r2']),
-                -np.mean(score['test_neg_mean_absolute_error']),
-                np.std(score['test_neg_mean_absolute_error'])),
-        elapsed_time)
+    scores = {
+        key: (f"{np.abs(np.mean(scores[f'test_{value}'])):.2f} +- "
+              f"{np.std(scores[f'test_{value}']):.2f}")
+        for key, value in scorers.items()
+    }
+
+    display = PredictionErrorDisplay(y_true=y, y_pred=y_pred, scores=scores)
```

Comment:
do we really want to directly use `PredictionErrorDisplay` in this example (which is not about advanced plotting stuff)? It seems contradictory to our recommendation to rather rely on `plot_prediction_error`

Reply:
Here I am doing this because we are computed the scores via cross-validation while the the `plot_*` function would compute for the set passed as argument. Another way would be to not call `plot_*` and then affect the legend to the axis manually.  I will make a try to see if this is readable